src/Simplify.elm

```diff
@@ -1041,9 +1229,42 @@ onlyErrors errors =
     }
 
 
+moduleContextToImportLookup : ModuleContext -> ImportLookup
+moduleContextToImportLookup context =
+    context.imports
+        |> Dict.map
+            (\moduleName import_ ->
+                case import_.exposed of
+                    ExposedAll ->
+                        import_
+
+                    ExposedSome some ->
+                        case Dict.get moduleName context.importedExposedVariants of
+                            Nothing ->
+                                import_
+
+                            Just importExposedVariants ->
+                                { import_
+                                    | exposed =
+                                        ExposedSome
+                                            (Set.union some importExposedVariants)
+                                }
+            )
+        |> Dict.insert context.moduleName
+            { alias = Nothing
+            , exposed =
+                ExposedSome
+                    (context.exposedVariants |> Dict.foldl (\_ -> Set.union) Set.empty)
+            }
+
+
 expressionVisitorHelp : Node Expression -> ModuleContext -> { errors : List (Error {}), rangesToIgnore : List Range, rightSidesOfPlusPlus : List Range, inferredConstants : List ( Range, Infer.Inferred ) }
 expressionVisitorHelp node context =
     let
+        importLookup : ImportLookup
+        importLookup =
+            moduleContextToImportLookup context
```

Comment:
This looks like a terribly expensive function to compute for every expression that we encounter.

I think it would be better to compute the `ImportLookup` once after we have visited all the imports. There is no visitor for that exact moment, so what we could do is compute it all in `fromProjectToModule` using the information `withFullAst`. We can then store an `ImportLookup` directly in the `ModuleContext` (I imagine, haven't fully checked).

Reply:
Thought about this as well – how to compute this once we finished going through all declarations... So why not do it at the end of [`withDeclarationListVisitor`](https://dark.elm.dmy.fr/packages/jfmengels/elm-review/latest/Review-Rule#withDeclarationListVisitor)?

Reply:
>Shall I merge the PR as it is, or do you want to fix that one first?

Lemme fix that first

Reply:
> So why not do it at the end of [withDeclarationListVisitor](https://dark.elm.dmy.fr/packages/jfmengels/elm-review/latest/Review-Rule#withDeclarationListVisitor)?

Yes, that works as well :+1: